src/internal/util/lift.ts

```diff
@@ -1,49 +1,30 @@
 /** @prettier */
 import { Observable } from '../Observable';
-import { Operator } from '../Operator';
 import { Subscriber } from '../Subscriber';
-import { TeardownLogic } from '../types';
+import { OperatorFunction, TeardownLogic } from '../types';
 
 /**
- * A utility to lift observables. Will also error if an observable is passed that does not
- * have the lift mechanism.
- *
- * We _must_ do this for version 7, because it is what allows subclassed observables to compose through
- * the operators that use this. That will be going away in v8.
- *
- * See https://github.com/ReactiveX/rxjs/issues/5571
- * and https://github.com/ReactiveX/rxjs/issues/5431
- *
- * @param source The source observable to lift
- * @param operator The operator to lift it with. Note that the operator possibly being undefined here
- * is related to issues around the "stanky" lifting of static creation functions as operators, see below.
+ * Used to determine if an object is an Observable with a lift function.
  */
-export function lift<T, R>(source: Observable<T>, operator?: Operator<T, R>): Observable<R> {
-  if (hasLift(source)) {
-    return source.lift(operator);
-  }
-  throw new TypeError('Unable to lift unknown Observable type');
+export function hasLift(source: any): source is { lift: InstanceType<typeof Observable>['lift'] } {
+  return source && typeof source.lift === 'function';
```

Comment:
```suggestion
  return typeof source?.lift === 'function';
```

Reply:
Oops... didn't see this and already did it per @cartant's suggestion.  Thanks @tonivj5!

Reply:
No problem!, I dind't see @cartant was faster than me.. hahah